### PR TITLE
fix: extract const in jsx

### PR DESCRIPTION
**Pull request by Jack-Works**
Before:

image

After:
image

### CODE REVIEW

**Comment by Jack-Works** [comment]
@sandersn hi it's a small fix plz have a look! 😆

File: src/services/refactors/extractSymbol.ts [hunk 1]
```diff
@@ -1100,7 +1100,12 @@ namespace ts.refactor.extractSymbol {
                     changeTracker.delete(context.file, node.parent);
                 }
                 else {
-                    const localReference = createIdentifier(localNameText);
+                    let localReference: Expression = createIdentifier(localNameText);
+                    // When extract to a new variable in JSX content, need to wrap a {} out of the new variable
+                    // or it will become a plain text
+                    if (isInJSXContent(node)) {
+                        localReference = createJsxExpression(void 0, localReference);
```
Comment (andrewbranch) [inline src/services/refactors/extractSymbol.ts:1107]:
```suggestion
                        localReference = createJsxExpression(/*dotDotDotToken*/ undefined, localReference);
```

File: src/services/refactors/extractSymbol.ts [hunk 2]
```diff
@@ -1112,6 +1117,12 @@ namespace ts.refactor.extractSymbol {
         const renameLocation = getRenameLocation(edits, renameFilename, localNameText, /*isDeclaredBeforeUse*/ true);
         return { renameFilename, renameLocation, edits };
 
+        function isInJSXContent(node: Node): node is JsxOpeningLikeElement {
```
Comment (andrewbranch) [inline src/services/refactors/extractSymbol.ts:1120]: Looks like this return type should be `node is JsxElement` if a type predicate is needed, but it doesn’t look like it’s needed.

```suggestion
        function isInJSXContent(node: Node) {
```

**Comment by andrewbranch** [comment]
Anyone else want to look before I merge?

**Pull request merged**